descartes_trajectory/include/descartes_trajectory/cartesian_interpolator.h

```diff
@@ -0,0 +1,121 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2014, Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTESIAN_INTERPOLATOR_H_
+#define CARTESIAN_INTERPOLATOR_H_
+
+#include <descartes_trajectory/cart_trajectory_pt.h>
+#include <tf/transform_datatypes.h>
+
+namespace descartes_trajectory
+{
+
+class TrajectorySegment
+{
+public:
+
+  virtual ~TrajectorySegment()
+  {
+
+  }
+
+  virtual bool interpolate(std::vector<descartes_core::TrajectoryPtPtr>& segment_points) = 0;
+
+protected:
+
+  TrajectorySegment(){}
+
+};
+
+class BlendSegment: public TrajectorySegment
+{
+public:
+
+  BlendSegment(const descartes_trajectory::CartTrajectoryPt& start,
+               const descartes_trajectory::CartTrajectoryPt& end,
+               double duration,
```

Comment:
The transitions should be replaced/removed.  This works makes them redundant.  The `TrajectoryPt` should capture the point velocity/timing at a minimum.  This would allow us to better prune our search tree.  I'm not sure how velocity generalizes to joint/cartesian points (maybe it doesn't have to).  Timing is pretty generic.